concent_api/middleman/queue_operations.py

```diff
@@ -162,6 +169,30 @@
         response_queue.task_done()
 
 
+async def is_authenticated(reader: StreamReader, writer: StreamWriter) -> bool:
+    challenge = create_random_challenge()
+
+    request_id = randint(1, 7777)
```

Comment:
This one looks suspicious. Shouldn't `request_id` be always unique? Maybe use the same method as for generating `request_id` for other frames. After all, authentication frames are frames too.

Reply:
actually, during the authentication phase it does not matter: it doesn't have to be unique, it has to be unique for short moment of simultaneous authentication. It is only to map the request with response - randomly choosing it from pretty wide range should be good enough.

Reply:
7777 does not seem like a very wide range.

Maybe this is the cause of #844?

Reply:
I think this is only for authentication case.